src/libs/OptionsListUtils.ts

```diff
@@ -248,14 +248,12 @@ Onyx.connect({
     },
 });
 
-const policyExpenseReports: OnyxCollection<Report> = {};
+let policyExpenseReports: OnyxCollection<Report> = {};
```

Comment:
or filter the reports in the `Onyx.connect` callback?

```ts
let policyExpenseReports: OnyxCollection<Report> = {};
Onyx.connect({
    key: ONYXKEYS.COLLECTION.REPORT,
    waitForCollectionCallback: true,
    callback: (allReports) => {
        policyExpenseReports = Object.entries(allReports)
            .filter(([reportID, report]) => ReportUtils.isPolicyExpenseChat(report))
            .reduce((memo, [reportID, report]) => {
                memo[reportID] = report;
                return memo;
            }, {});
    }
});
```

Reply:
note that in this snippet ^ I'm avoiding using a spread in a reduce, which is very inefficient, especially for a potentially large object

Reply:
@roryabraham @shubham1206agra 

On a second thought, I think we should remove the `Onyx.connect` here. The reason is that it's only used to populate `policyExpenseReports` which is only used in `getPolicyExpenseReportOption`. I looked for the usages of this function throughout the codebase and it turns out that the `report` we pass to this function is already verified as a policy expense report, so it is not ideal to get the report again from the collection we store from `Onyx.connect` and then access the report from collection by using the `reportID` passed by the report object to the `getPolicyExpenseReportOption`.

<img width="1728" alt="Screenshot 2024-03-20 at 5 03 57 PM" src="https://github.com/Expensify/App/assets/47336142/cbcf0802-59b5-4ba5-b45c-d4e6ed6229cc">


<br/>
<hr/>
Below is how the changes will look like:
<br/>

```diff
diff --git a/src/libs/OptionsListUtils.ts b/src/libs/OptionsListUtils.ts
index 73620a2b6f..ab132c21c2 100644
--- a/src/libs/OptionsListUtils.ts
+++ b/src/libs/OptionsListUtils.ts
@@ -248,15 +248,6 @@ Onyx.connect({
     },
 });
 
-let policyExpenseReports: OnyxCollection<Report> = {};
-Onyx.connect({
-    key: ONYXKEYS.COLLECTION.REPORT,
-    waitForCollectionCallback: true,
-    callback: (report) => {
-        policyExpenseReports = report;
-    },
-});
-
 let allTransactions: OnyxCollection<Transaction> = {};
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.TRANSACTION,
@@ -735,12 +726,10 @@ function createOption(
  * Get the option for a policy expense report.
  */
 function getPolicyExpenseReportOption(report: Report): ReportUtils.OptionData {
-    const expenseReport = policyExpenseReports?.[`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`];
-
     const option = createOption(
-        expenseReport?.visibleChatMemberAccountIDs ?? [],
+        report?.visibleChatMemberAccountIDs ?? [],
         allPersonalDetails ?? {},
-        expenseReport ?? null,
+        report,
         {},
         {
             showChatPreviewLine: false,
@@ -749,7 +738,7 @@ function getPolicyExpenseReportOption(report: Report): ReportUtils.OptionData {
     );
 
     // Update text & alternateText because createOption returns workspace name only if report is owned by the user
-    option.text = ReportUtils.getPolicyName(expenseReport);
+    option.text = ReportUtils.getPolicyName(report);
     option.alternateText = Localize.translateLocal('workspace.common.workspace');
     option.selected = report.selected;
     option.isSelected = report.selected;

```

Reply:
sounds good. If we can add some inline local validation of the report object with `ReportUtils.isPolicyExpenseChat`, that could be a good safety measure without any significant performance cost